File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/EOnSecondaryContainerLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EOnSecondaryContainerLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl28
            AppTheme.STANDARD -> R.color.cstl28
            AppTheme.BLUE -> R.color.cbl28
            AppTheme.GREEN -> R.color.cgl28
            AppTheme.RED -> R.color.crl28
            AppTheme.NOTHING -> R.color.cnol28
        }
    }
}

File: lib/layer2_core/light/e_on_secondary_container_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class EOnSecondaryContainerLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl28;
      case 'standard': return cstl28;
      case 'blue': return cbl28;
      case 'green': return cgl28;
      case 'red': return crl28;
      case 'nothing': return cnol28;
      default: return cstl28;
    }
  }
}
